packages/arb-avm-cpp/data_storage/src/value/machine.cpp

```diff
@@ -247,9 +243,16 @@ SaveResults saveMachine(ReadWriteTransaction& transaction,
         return save_res;
     }
 
-    auto status = saveMachineState(transaction, machine);
-    if (!status.ok()) {
-        return {0, status};
+    auto machine_save_res = saveMachineState(transaction, machine);
+    if (!machine_save_res.first.ok()) {
+        return {0, machine_save_res.first};
+    }
+    auto core_code = dynamic_cast<CoreCode*>(machine.machine_state.code.get());
```

Comment:
It seems to me like this wouldn't contain a CoreCode in practice. Is this function called anywhere? Tests maybe? Maybe rename it to saveTestMachine if that's the only time it's usable.

Reply:
Renamed this method as suggested